docs/posts/ibis-bench/index.qmd

```diff
@@ -0,0 +1,1103 @@
+    )
+    .when(
+        ibis._["instance_type"].startswith("c3"),
+        "Intel Sapphire Rapids",
+    )
+    .when(
+        ibis._["instance_type"] == "work laptop",
+        "Apple M1 Max",
+    )
+    .when(
+        ibis._["instance_type"] == "personal laptop",
+        "Apple M2 Max",
+    )
+    .else_("unknown")
+    .end()
+)
+cpu_num_cases = (
+    ibis.case()
+    .when(
+        ibis._["instance_type"].contains("-"),
+        ibis._["instance_type"].split("-")[-1].cast("int"),
+    )
+    .when(ibis._["instance_type"].contains("laptop"), 12)
+    .else_(0)
+    .end()
+)
+memory_gb_cases = (
+    ibis.case()
+    .when(
+        ibis._["instance_type"].contains("-"),
+        ibis._["instance_type"].split("-")[-1].cast("int") * 4,
+    )
+    .when(ibis._["instance_type"] == "work laptop", 32)
+    .when(ibis._["instance_type"] == "personal laptop", 96)
+    .else_(0)
+    .end()
+)
+
+instance_details = (
+    t.group_by("instance_type")
+    .agg()
+    .mutate(
+        cpu_type=cpu_type_cases, cpu_cores=cpu_num_cases, memory_gbs=memory_gb_cases
+    )
+).order_by("memory_gbs", "cpu_cores", "instance_type")
+
+cpu_types = sorted(
+    instance_details.distinct(on="cpu_type")["cpu_type"].to_pyarrow().to_pylist()
+)
+
+instance_details
+```
+
+### What's in the data?
+
+With the data, we can see we ran the benchmark on scale factors:
+
+```{python}
+sfs = sorted(t.distinct(on="sf")["sf"].to_pyarrow().to_pylist())
+sfs
+```
+
+:::{.callout-note title="What is a scale factor?" collapse="true"}
+A scale factor is roughly the size of the data in memory in gigabytes. For
+example, a scale factor of 1 means the data is roughly 1GB in memory.
+
+Stored on disk in Parquet format, the data is smaller -- about 0.38GB for scale
+factor 1 with the compression settings used in this benchmark.
+:::
+
+We can look at the total execution time by scale factor:
+
+```{python}
+#| code-fold: true
+#| code-summary: "Show bar plot code"
+c = px.bar(
+    t.group_by("sf").agg(total_seconds=t["execution_seconds"].sum()),
+    x="sf",
+    y="total_seconds",
+    category_orders={"sf": sfs},
+    title="total execution time by scale factor",
+)
+c
+```
+
+We ran on the following systems:
+
+```{python}
+systems = sorted(t.distinct(on="system")["system"].to_pyarrow().to_pylist())
+systems
+```
+
+:::{.callout-note title="What is a system?" collapse="true"}
+For convenience in this benchmark, a 'system' is defined as a hyphen-separated
+naming convention where:
+
+- `ibis-*`: Ibis was used
+    - `ibis-<backend>`: Ibis dataframe code was used with the given backend
+    - `ibis-<backend>-sql`: SQL code was used via Ibis on the given backend
+- `polars-*`: Polars was used
+    - `polars-lazy`: Polars was used with the LazyFrames API
+:::
+
+We can look at the total execution time by system:
+
+```{python}
+#| code-fold: true
+#| code-summary: "Show bar plot code"
+c = px.bar(
+    t.group_by("system").agg(total_seconds=t["execution_seconds"].sum()),
```

Comment:
Not entirely sure how useful this is given that polars is missing a large number of queries in some cases. If you're going to keep this I this should be removed or replaced with something else that's try to convey the point you're trying to make here.

It looks like you're using total execution time for time measurements throughout. I would suggest something that accounts for the fact that polars couldn't run most of the queries at larger scale factors on less powerful hardware.